### PR TITLE
Add number of task to the URL in submission grid row link

### PR DESCRIPTION
**Pull request by msmner**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/479

### CODE REVIEW

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx [hunk 1]
```diff
@@ -32,6 +32,11 @@ const SubmissionGridRow = ({ submission }: ISubmissionGridRowProps) => {
         isOfficial,
     } = submission;
 
+    const getProblemNumber = useMemo(
+        () => problemName.substring(0, 2).substring(1),
```
Comment (mishogv) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx:36]: magic numbers

**Review by mishogv — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx [hunk 2]
```diff
@@ -32,6 +32,11 @@ const SubmissionGridRow = ({ submission }: ISubmissionGridRowProps) => {
         isOfficial,
     } = submission;
 
+    const getProblemNumber = useMemo(
```
Comment (Minkov) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx:35]: `getProblemNumber` shows behavior, but it a state..

**Review by Minkov — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx (hunk 1 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx]: I removed the method and instead added a property OrderBy to the problem, which gives me its number.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx (hunk 2 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx]: Function removed.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx [hunk 3]
```diff
@@ -84,10 +85,9 @@ const SubmissionGridRow = ({ submission }: ISubmissionGridRowProps) => {
             </div>
             <div className={styles.detailsContainer}>
                 <div>
-                    {/* TODO: Fix this URL once https://github.com/SoftUni-Internal/exam-systems-issues/issues/184 is done */}
                     <LinkButton
                       text={problemName}
-                      to={`/contests/${contestId}/${participationType}`}
+                      to={`/contests/${contestId}/${participationType}#${orderBy + 1}`}
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx:90]: Is this correct? Order by is a number which can be configured in the administration and sometimes administrators put 10, 20, 30 for tasks 1, 2, 3 so it's easier to put a task between 10 and 20 by setting 11 as a value. Doesn't this mean that the url should contain the plain order by value?

**Review by shsimeonova — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx (hunk 3 above)
Comment (msmner) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-grid-row/SubmissionGridRow.tsx:90]: Their orderBy is normalized and then recalculated by this function in useProblems() hook, so it should be ok: 

const normalizeOrderBy = (problems: IProblemType[]) => {
    problems.sort((x, y) => x.orderBy - y.orderBy);

    return problems.map((p, i) => ({
        ...p,
        orderBy: i + 1,
    }));
};

**Review by mishogv — APPROVED** [review]

**Review by mishogv — APPROVED** [review]

**Review by shsimeonova — APPROVED** [review]

**Review by Minkov — APPROVED** [review]

**Pull request merged**